andela/wg-rogue-one
# #157731625: Authors field for exercises should be a foreign key

### What does this PR do?
This PR enables the authors to be a select field in the view and a foreign key in the Exercises
#### Description of Task to be completed?
Previously, wger presented the author's field as an optional single field, this didn't provide enough context to who created the exercise.

This PR allows an exercise to pick from a list of authors or by default resolve to wger if the exercise was explicitely created by wger.

#### Screenshots
<img width="1440" alt="info_author" src="https://user-images.githubusercontent.com/17830204/41906809-9b8c3ff2-7947-11e8-9237-5f48c53e6bcc.png">
Create an exercise

<img width="1440" alt="add_author_2" src="https://user-images.githubusercontent.com/17830204/41906828-aa9dff44-7947-11e8-8566-71154e39c315.png">
Select author from author list

<img width="1440" alt="add_author_1" src="https://user-images.githubusercontent.com/17830204/41906844-b76a4430-7947-11e8-842a-ea02eb0d6cc6.png">
On the right hand side we can see the author

#### How should this be manually tested?
1. Follow the readme.rst procedures on how to set up the application locally 
    NB: Make sure you run the migrations command
2. Login into the system
3. From the navigation bar, select exercises link and either select exercises or add new exercise.
4. In the form, the licensed authors' field can invoke a list of authors.
5. Select an author or otherwise, it will resolve to the default value.
6. From created exercise view, on the right-hand side, the author will be created.


#157731625
